### PR TITLE
Layout add user screen and disable user dialog

### LINKED ISSUE #329: Add user screen and disable user dialog
- [x] create new route for add user
- [x] click `add user` in command bar goes to add user page
- [x] create form with add user button (PrimaryButton)

- Data fetching:
  - [x] query for app config
  - [x] add type def for password policy config
  - [x] show email, username, phone fields depending on app config -> login id
  - [x] show password field depending on app config -> authenticator whether password option is chosen in primary authenticator

- Password field component:
  - [x] install zxcvbn
  - [x] create password strength meter
  - [x] policy list, with `passed` style
  - [x] policy list check

- Validation: 
  - [x] check password against password policy on save
  - [x] check if one and only one of available fields in email, username, phone filled

- [x] block navigation if form dirty
- [x] create disable user dialog
- all button dummy at the moment

### PR DESCRIPTION
**Pull request by axolotl0212**
Closes #329

Summary:
password field + strength meter + policy list
query app config
layout add user screen
validation on save
add action column in user list
disable user button in action column
disable user dialog

Add user, disable user mutation integration to be done

### CODE REVIEW

File: portal/src/graphql/adminapi/UsersList.tsx [hunk 1]
```diff
@@ -158,11 +177,56 @@ const PlainUsersList: React.FC<PlainUsersListProps> = function PlainUsersList(
...
+    },
+    []
+  );
+
+  const onRenderUserItemColumn = useCallback(
+    (item: UserListItem, _index?: number, column?: IColumn) => {
+      switch (column?.key) {
+        case "action":
+          return (
+            <ActionButton
```
Comment (louischan-oursky) [inline portal/src/graphql/adminapi/UsersList.tsx:199]: The button text is black. Can we add a theme to make it more aligned with the design?

Comment (axolotl0212) [reply portal/src/graphql/adminapi/UsersList.tsx:199]: Added a theme to update the button text colour, but the text colour is the same as hover text colour

**Comment by axolotl0212** [comment]
updated and rebased

File: portal/src/PasswordField.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      />
+      <section className={styles.passwordPolicy}>
+        {passwordPolicyData.map((policy) => (
+          <Text
+            key={policy.messageId}
+            className={cn(styles.policyLine, {
+              [styles.policySatisfied]: isPasswordPolicySatisfied[policy.key],
+            })}
+          >
+            {"• "}
```
Comment (kiootic) [inline portal/src/PasswordField.tsx:228]: Bullet items should be displayed using `ul` & `li`.

File: portal/src/graphql/adminapi/DisableUserDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+} from "@fluentui/react";
+import { Context, FormattedMessage } from "@oursky/react-messageformat";
+
+interface DisableUserDialogProps {
+  onDismiss: () => void;
+  userID: string;
+  username: string | null;
+}
+
+const dialogStyles = { main: { minWidth: "400px !important", minHeight: 0 } };
```
Comment (kiootic) [inline portal/src/graphql/adminapi/DisableUserDialog.tsx:17]: I suppose we can use the default styling ([official recommended](https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog) not to exceed 340px in width).

**Comment by axolotl0212** [comment]
@kiootic updated, used ul, li, and remove custom width style in disable user dialog

File: portal/src/graphql/adminapi/AddUserScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    const loginIdKeys = appConfig?.identity?.login_id?.keys ?? [];
+    // We consider them as enabled if they are listed as allowed login ID keys.
+    const username = loginIdKeys.find((key) => key.type === "username") != null;
+    const email = loginIdKeys.find((key) => key.type === "email") != null;
+    const phone = loginIdKeys.find((key) => key.type === "phone") != null;
+
+    const primaryAuthenticators =
+      appConfig?.authentication?.primary_authenticators ?? [];
+    const password =
+      primaryAuthenticators.length > 0
```
Comment (kiootic) [inline portal/src/graphql/adminapi/AddUserScreen.tsx:99]: @axolotl0212 sorry the condition is actually quite complex:
- if only `password` in `primary_authenticators`, all three login ID type need password.
- if only `oob_otp` in `primary_authenticators`, only email & phone is possible, and no password is needed.
- if `primary_authenticators` has both `oob_otp` & `password`, and `oob_otp` is before `password`, email & phone does not need password, but username need password.
- if `primary_authenticators` has both `oob_otp` & `password`, and `oob_otp` is after `password`, all three login ID type need password.

This need design update, so we can keep it as-is for now.

cc @louischan-oursky @keithtkl

File: portal/src/PasswordField.tsx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+      <ul className={styles.passwordPolicy}>
+        {passwordPolicyData.map((policy) => (
+          <li
+            key={policy.messageId}
+            className={cn(styles.policyLine, {
+              [styles.policySatisfied]: isPasswordPolicySatisfied[policy.key],
+            })}
+          >
+            <Text>
+              {"• "}
```
Comment (kiootic) [inline portal/src/PasswordField.tsx:229]: Please use CSS to display the bullet. The bullet icon should not be included as part of content: it is a pure styling/display concern.

Comment (axolotl0212) [reply portal/src/PasswordField.tsx:229]: Thanks, didn't notice that we can add bullet point by CSS

**Comment by axolotl0212** [comment]
rebased and updated

**Review by kiootic — APPROVED** [review]

**Review by louischan-oursky — APPROVED** [review]

File: portal/src/graphql/adminapi/AddUserScreen.tsx (hunk 4 above)
Comment (keithtkl) [reply portal/src/graphql/adminapi/AddUserScreen.tsx:99]: @kiootic @louischan-oursky What do you think if we simply take out password field for the case that if only `oob_otp` in `primary_authenticators` while other cases still requires user to input the password when add an user?

Comment (louischan-oursky) [reply portal/src/graphql/adminapi/AddUserScreen.tsx:99]: I guess we would disable the password field in cases where password is not needed.

**Pull request merged**